files/runScanner.sh

```diff
@@ -34,59 +43,76 @@ sed -i 's/^ip_address=.*//' /opt/brother/scanner/brscan-skey/brscan-skey.config
 if [[ -z "$HOST_IPADDRESS" ]]; then
 	echo "no host IP configured, using default discovery"
 else
+	echo "using host IP: $HOST_IPADDRESS"
 	echo "ip_address=$HOST_IPADDRESS" >> /opt/brother/scanner/brscan-skey/brscan-skey.config
 fi
 echo "-----"
 
+echo "setting up username:"
+sed -i 's/^user=.*//' /opt/brother/scanner/brscan-skey/brscan-skey.config
+echo "user=${USERNAME:-docker}" >> /opt/brother/scanner/brscan-skey/brscan-skey.config
+echo "using username: ${USERNAME:-docker}"
+echo "-----"
+
 echo "whole config:"
 cat /opt/brother/scanner/brscan-skey/brscan-skey.config
 echo "-----"
 
 echo "starting scanner drivers..."
-su - $USERNAME -c "/usr/bin/brsaneconfig4 -a name=$NAME model=$MODEL ip=$IPADDRESS"
-su - $USERNAME -c "/usr/bin/brscan-skey"
+driver_cmds=("/usr/bin/brsaneconfig4 -a name=$NAME model=$MODEL ip=$IPADDRESS" /usr/bin/brscan-skey)
+for cmd in "${driver_cmds[@]}"
+do
+	#check if failed
+	$cmd
+	if [ $? -ne 0 ]; then
+		echo "failed to start: `$cmd`"
+		exit 1
+	fi
+done
+echo "-----"
+
+echo "setting up script permissions..."
+#change ownership of scripts to root and setuid
+chown root /opt/brother/scanner/brscan-skey/script/*.sh
+chmod u+s /opt/brother/scanner/brscan-skey/script/*.sh
```

Comment:
Is this the solution to my comment above on permissions? Basically makes the script run as root always, correct? Wouldn't it be cleaner to set a proper UID then running this as root? Especially with the o+x in the next line, basically everyone can run this script as root. Technically shouldn't be an issue, but does not feel "clean"...

Reply:
that kinda depends on a users own script. your call

Reply:
e.g. have a button to create a pdf for user 1001:1001 and another for 2022:2022, only root is able to change a file owner